Lib/fractions.py

```diff
@@ -183,7 +183,7 @@ class Fraction(numbers.Rational):
     __slots__ = ('_numerator', '_denominator')
 
     # We're immutable, so use __new__ not __init__
-    def __new__(cls, numerator=0, denominator=None, *, _normalize=True):
+    def __new__(cls, numerator=0, denominator=None):
```

Comment:
I'm wondering whether we'd need to keep the `_normalize` argument and issue a deprecation warning. It's true that it's undocumented, but I suspect that Hyrum's law applies here - it wouldn't be at all surprising if there are people passing `_normalize=False` for speed reasons.

Reply:
PEP 8 documents `_single_leading_underscore`-style names as "internal use" indicator.  No news (after 3c286e2e0d), no documentation...

So, I think it's not a part of the Fraction API and we can break it.

>  it wouldn't be at all surprising if there are people passing _normalize=False for speed reasons.

Indeed, [they do](https://github.com/search?l=Python&p=1&q=Fraction%28+_normalize%3DFalse&type=Code)  :(

Probably, there will be no harm to expose this kwarg as `normalize`, but I'm not sure if we should also issue a deprecation warning.  I'll revert this part for now.

Reply:
> PEP 8 documents [...]

I think you can and should assume that all core developers are very familiar with PEP 8. :-)

> So, I think it's not a part of the Fraction API and we can break it.

That's a rather dogmatic approach. On CPython, we should prefer a more pragmatic approach: if we have reasonable grounds to believe that a change might break user code then we should take reasonable steps to avoid that breakage. It's not friendly to knowingly break existing code and then tell people "well, it was your fault for using something undocumented".